Dockerfile

```diff
@@ -0,0 +1,22 @@
+# Builder
+FROM node:alpine as builder
+
+RUN apk add --no-cache git
+
+WORKDIR /src
+
+COPY package.json /src/package.json
+RUN npm install
+
+COPY . /src
+RUN npm run build
+
+
+# App
+FROM nginx:latest
```

Comment:
Wouldn't it be better to have two Dockerfiles present to use Node's server for running development version of Riot and nginx for quick deployment? It'd make even the development easier because if you mount the source folder and run the server, Node could automatically recompile your source the same way React does.

Also, perhaps instead of using `node:alpine` you might want to use *just* the alpine itself or even *nginx*, then have a shell script (or python or whatever) and do this:

```shell
# build.sh
#!/bin/sh
set -xe
apk add node
apk add npm  # or yarn
yarn install
yarn run build
apk remove npm
apk remove node
# clean cache and everything silly
```

Then in the Dockerfile:

```Dockerfile
RUN build.sh
```

which would in the end mean that you could build on a single filesystem layer which would mean smaller image size and less unnecessary bloatware.

Also: https://github.com/oracle/smith for hammering the size even more (but it's more advanced than building a proper script).

Reply:
Using the docker image for development doesn't give us much gain here. Node already does a great job of compartmentalizing dependencies and is lightweight enough to run on a local machine without the need for Docker. The concerns that Docker development images would normally solve are more issues with webpack and developer workflow, which I think are best solved on a case-by-case basis.